src/tv2-common/helpers/graphics/pilot/index.ts

```diff
@@ -217,46 +194,52 @@ export class PilotGraphicGenerator {
 		}
 	}
 
-	private createContent(): WithTimeline<GraphicsContent> {
-		if (this.config.studio.GraphicsType === 'HTML') {
-			return GetPilotGraphicContentCaspar(this.config, this.context, this.parsedCue, this.settings.caspar, this.engine)
-		} else {
-			return GetPilotGraphicContentViz(
-				this.config,
-				this.context,
-				this.settings.viz,
-				this.parsedCue,
-				this.engine,
-				this.adlib
-			)
-		}
-	}
-
-	private getPrerollDuration(): number {
+	protected getPrerollDuration(): number {
 		return this.config.studio.GraphicsType === 'HTML'
 			? this.config.studio.CasparPrerollDuration
 			: this.config.studio.VizPilotGraphics.PrerollDuration
 	}
 
-	private getSourceLayer(): SharedSourceLayers {
-		const engine = this.engine
-		return IsTargetingWall(engine)
-			? SharedSourceLayers.WallGraphics
-			: IsTargetingTLF(engine)
-			? SharedSourceLayers.PgmGraphicsTLF
-			: IsTargetingOVL(engine)
-			? SharedSourceLayers.PgmPilotOverlay
-			: SharedSourceLayers.PgmPilot
+	protected getSourceLayer(): SharedSourceLayers {
+		switch (this.engine) {
+			case 'WALL':
+				return SharedSourceLayers.WallGraphics
+			case 'TLF':
+				return SharedSourceLayers.PgmGraphicsTLF
+			case 'OVL':
+				return SharedSourceLayers.PgmPilotOverlay
+			case 'FULL':
+				return SharedSourceLayers.PgmPilot
+			default:
+				assertUnreachable(this.engine)
+		}
 	}
 
-	private getOutputLayer() {
-		const engine = this.engine
-		return IsTargetingWall(engine)
-			? SharedOutputLayers.SEC
-			: IsTargetingOVL(engine)
-			? SharedOutputLayers.OVERLAY
-			: IsTargetingFull(engine)
-			? SharedOutputLayers.PGM
-			: SharedOutputLayers.OVERLAY
+	protected getOutputLayer(): SharedOutputLayers {
+		switch (this.engine) {
+			case 'WALL':
+				return SharedOutputLayers.SEC
+			case 'OVL':
+				return SharedOutputLayers.OVERLAY
+			case 'FULL':
+			case 'TLF':
+				return SharedOutputLayers.PGM
```

Comment:
note to reviewers: this might look incorrect but `IsTargetingFull` returns true for 'TLF'

Reply:
Is there some way we can make this "hidden" coupling less hidden? It seems like it would be easy to sometime down the road decide that "TLF" should no longer target full and then it won't be as clear that we need to change something here.

I don't have a good suggestion except for something like having this conditional be placed close to the `isTargetingFull` logic.

Reply:
I don't like the `TLF` target at all. It does not seem to be representing any logical or physical graphics engines/channels, and its pieces are exclusive with `FULL` anyway. According to what I've learned when implementing [this fix](https://github.com/tv2/sofie-blueprints-inews/pull/111), the only place where there might be an actual difference is parsing (pairing targets), but other than that it just causes confusion.